src/plugins/dashboard/public/dashboard_api/settings_manager.ts

```diff
@@ -0,0 +1,140 @@
+}) {
+  const syncColors$ = new BehaviorSubject<boolean>(
+    initialState?.syncColors ?? DEFAULT_DASHBOARD_INPUT.syncColors
+  );
+  function setSyncColors(syncColors: boolean) {
+    if (syncColors !== syncColors$.value) syncColors$.next(syncColors);
+  }
+  const syncCursor$ = new BehaviorSubject<boolean>(
+    initialState?.syncCursor ?? DEFAULT_DASHBOARD_INPUT.syncCursor
+  );
+  function setSyncCursor(syncCursor: boolean) {
+    if (syncCursor !== syncCursor$.value) syncCursor$.next(syncCursor);
+  }
+  const syncTooltips$ = new BehaviorSubject<boolean>(
+    initialState?.syncTooltips ?? DEFAULT_DASHBOARD_INPUT.syncTooltips
+  );
+  function setSyncTooltips(syncTooltips: boolean) {
+    if (syncTooltips !== syncTooltips$.value) syncTooltips$.next(syncTooltips);
+  }
+  const tags$ = new BehaviorSubject<string[]>(initialState?.tags ?? DEFAULT_DASHBOARD_INPUT.tags);
+  function setTags(tags: string[]) {
+    if (!fastIsEqual(tags, tags$.value)) tags$.next(tags);
+  }
+  const titleManager = initializeTitles(initialState ?? {});
+  const useMargins$ = new BehaviorSubject<boolean>(
+    initialState?.useMargins ?? DEFAULT_DASHBOARD_INPUT.useMargins
+  );
+  function setUseMargins(useMargins: boolean) {
+    if (useMargins !== useMargins$.value) useMargins$.next(useMargins);
+  }
+
+  function getSettings() {
+    return {
+      ...titleManager.serializeTitles(),
+      syncColors: syncColors$.value,
+      syncCursor: syncCursor$.value,
+      syncTooltips: syncTooltips$.value,
+      tags: tags$.value,
+      timeRestore: timeRestore$.value,
+      useMargins: useMargins$.value,
+    };
+  }
+
+  function setSettings(settings: DashboardStateFromSettingsFlyout) {
+    setSyncColors(settings.syncColors);
+    setSyncCursor(settings.syncCursor);
+    setSyncTooltips(settings.syncTooltips);
+    setTags(settings.tags);
+    setTimeRestore(settings.timeRestore);
+    setUseMargins(settings.useMargins);
+    titleManager.titlesApi.setHidePanelTitle(settings.hidePanelTitles);
+    titleManager.titlesApi.setPanelDescription(settings.description);
+    titleManager.titlesApi.setPanelTitle(settings.title);
+  }
+
+  return {
+    api: {
+      ...titleManager.titlesApi,
+      getSettings,
+      settings: {
+        syncColors$,
+        syncCursor$,
+        syncTooltips$,
+        useMargins$,
+      },
+      setSettings,
+      setTags,
+      timeRestore$,
+    },
+    comparators: {
+      ...titleManager.titleComparators,
+      syncColors: [syncColors$, setSyncColors],
+      syncCursor: [syncCursor$, setSyncCursor],
+      syncTooltips: [syncTooltips$, setSyncTooltips],
+      useMargins: [useMargins$, setUseMargins],
+    } as StateComparators<
+      Pick<
+        DashboardState,
+        | 'description'
+        | 'hidePanelTitles'
+        | 'syncColors'
+        | 'syncCursor'
+        | 'syncTooltips'
+        | 'title'
+        | 'useMargins'
+      >
+    >,
+    internalApi: {
+      getState: (): Pick<
+        DashboardState,
```

Comment:
Using pick because types are from `DashboardState`. The fact that `DashboardState` gets these types from `DashboardOptions` is an implementation detail of `DashboardState` that should not be exposed for this usage.

Reply:
Any particular reason why you want to keep them separate? As I see it, the `settingsManager` manages state of type `DashboardOptions` and it's none of its concern that the `DashboardOptions` will eventually comprise part of `DashboardState`.

This is mostly pedantic, so no need to change anything - just curious!

Reply:
`settingsManager` manages more then `DashboardOptions`. For example, `settingsManager` manages 'description' and 'title', which are not part of `DashboardOptions`.

Reply:
Ah, that makes sense - should've looked harder at the code.